src/main/kotlin/org/malachite/estella/aplication/api/ApplicationController.kt

```diff
@@ -22,47 +30,58 @@ class ApplicationController(
-        val jobSeeker = securityService.getJobSeekerFromJWT(jwt) ?: return ResponseEntity.status(404).body("Unauthenticated")
-        val saved = applicationService.insertApplicationLoggedInUser(offerId,jobSeeker, applicationPayload)
-        return ResponseEntity.created(URI("/api/applications/" + saved.id)).build()
-    }
+    fun applyForAnOffer(
+        @PathVariable offerId: Int, @CookieValue("jwt") jwt: String?,
+        @RequestBody applicationPayload: ApplicationLoggedInPayload
+    ) =
+        securityService.getJobSeekerFromJWT(jwt)
+            ?.let { applicationService.insertApplicationLoggedInUser(offerId, it, applicationPayload) }
+            ?.let { CREATED(it) }
+            ?: UNAUTH
 
     @CrossOrigin
     @PostMapping("/apply/{offerId}/no-user")
-    fun applyForAnOffer(@PathVariable offerId: Int, @RequestBody applicationPayload: ApplicationNoUserPayload): SuccessMessage =
-        applicationService.insertApplicationWithoutUser(offerId, applicationPayload).let { SuccessMessage }
+    fun applyForAnOffer(
+        @PathVariable offerId: Int,
+        @RequestBody applicationPayload: ApplicationNoUserPayload
+    ) = applicationService
+        .insertApplicationWithoutUser(offerId, applicationPayload)
+        .let { CREATED(it) }
 
 
     @CrossOrigin
     @GetMapping("/{applicationId}")
-    fun getApplicationById(@PathVariable applicationId: Int) =
-        applicationService.getApplicationById(applicationId)
+    fun getApplicationById(@PathVariable applicationId: Int): ResponseEntity<ApplicationDTO> =
+        applicationService.getApplicationById(applicationId).let { ResponseEntity.ok(it) }
 
     @CrossOrigin
     @GetMapping("/")
-    fun getAllApplications() =
+    fun getAllApplications(): ResponseEntity<List<ApplicationDTO>> =
         applicationService.getAllApplications()
+            .let { ResponseEntity.ok(it) }
 
     @CrossOrigin
     @GetMapping("/offer/{offerId}")
-    fun getAllApplicationsByOffer(@PathVariable offerId: Int) =
+    fun getAllApplicationsByOffer(@PathVariable offerId: Int): ResponseEntity<List<ApplicationDTO>> =
         applicationService.getApplicationsByOffer(offerId)
+            .let { ResponseEntity.ok(it) }
 
     @CrossOrigin
     @GetMapping("/job-seeker/{jobSeekerId}")
-    fun getAllApplicationsByJobSeeker(@PathVariable jobSeekerId: Int) =
-            applicationService.getApplicationsByJobSeeker(jobSeekerId)
+    fun getAllApplicationsByJobSeeker(@PathVariable jobSeekerId: Int): ResponseEntity<List<ApplicationDTO>> =
+        applicationService.getApplicationsByJobSeeker(jobSeekerId)
+            .let { ResponseEntity.ok(it) }
 
     @CrossOrigin
-    @DeleteMapping("/delete/{applicationId}")
-    fun deleteApplication(@PathVariable applicationId: Int) =
+    @DeleteMapping("/{applicationId}")
+    fun deleteApplication(@PathVariable applicationId: Int): ResponseEntity<Any> =
```

Comment:
Teoretycznie też możesz usunąć ten zwracany typ, ale to jak nic innego nie będzie, to raczej nie wadzi

Reply:
Dobra mimo wszystko zostawię te typy, bo jakoś mi się lepiej je czyta jak są XD